Sources/Basics/FileSystem/AsyncLocalFileSystem.swift

```diff
@@ -0,0 +1,266 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+#if swift(>=5.5.2)
+import Foundation
+import TSCBasic
+
+/// Concrete FileSystem implementation which communicates with the local file system.
+private actor LocalFileSystem: AsyncFileSystem {
```

Comment:
Should we instead have a generic `AsyncFileSystem` that can wrap any `FileSystem` and make it async? Clients can provide an arbitrary `FileSystem` implementation to the `Workspace` initializer and I would expect that to be used throughout, so almost nothing would be able to use `AsyncFileSystem` if we want to keep that contract, right?

Reply:
That's a great idea, let me try that out! 👍

Reply:
Probably not generic, but owning an existential `FileSystem`? Otherwise we'll have to make types owning `AsyncFileSystem` generic. `FileSystem` is already passed around in a boxed existential form everywhere anyway.

Reply:
Sorry, I meant "generic" in the english sense, not using actual generics in Swift :)